### PR TITLE
Added details to the ACL documentation.

### PR DESCRIPTION
**Pull request by matejak**
I did my best to expand the ACL documentation to make a story of sharing mailboxes somehow smoother for people who are new to dovecot like me.

- Made it clear that the per-user and global ACLs serve different purposes, and aren't different implementation of the same.
- Added an interpretation to the global ACL example.
- Connected ACL documentation to the dictionaries documentation, which was only present as hints in config files or external tutorials.

based on the thread https://dovecot.org/pipermail/dovecot/2020-September/119925.html Thanks to Daniel Miller!

### CODE REVIEW

File: source/settings/plugin/acl.rst [hunk 1]
```diff
@@ -254,3 +263,18 @@ List cache
...
+^^^^^^^^^^^^
+
+In order for an ACL to be fully useful, it has to be communicated to IMAP clients.
+For example, if you use ACL to share a mailbox to another user, the client has to
+be explicitly told to check out the other user's mailbox too, as that one is shared.
+Placing the ACL file makes the ACL effective, but ``dovecot`` doesn't take care of
+the user->shared mailboxes mapping out of the box, and as a result, it won't publish
+shared mailboxes to clients if this is not set up.
+You have to configure this manually by defining an appropriate
+`dictionary <https://wiki.dovecot.org/Dictionary>` to store the share map.
```
Comment (cmouse) [inline source/settings/plugin/acl.rst:277]: needs `_` at the end.

File: source/settings/plugin/acl.rst [hunk 2]
```diff
@@ -42,6 +42,9 @@ ACLs can be enabled in ``dovecot.conf`` with:
     #acl_globals_only = yes
   }
 
+Global and per-user ACLs have different use cases, and both can be enabled at
```
Comment (cmouse) [inline source/settings/plugin/acl.rst:45]: this paragraph seems bit random here.

**Review by cmouse — CHANGES_REQUESTED** [review]

**Comment by cmouse** [comment]
The commit message is also not in line with the normal way we write commit messages, see log for details.

File: source/settings/plugin/acl.rst (hunk 2 above)
Comment (matejak) [reply source/settings/plugin/acl.rst:45]: I can relate to that remark, but when I first read through the text, I thought that those two are in an either/or relation. I don't insist on keeping that sentence there, but perhaps you can suggest a better wording?

File: source/settings/plugin/acl.rst (hunk 1 above)
Comment (matejak) [reply source/settings/plugin/acl.rst]: Fixed, along with the other place where I copied this from.

**Comment by matejak** [comment]
Ping, is there something I can do to get this closer to being merged?

File: source/settings/plugin/acl.rst (hunk 2 above)
Comment (cmouse) [reply source/settings/plugin/acl.rst:45]: Maybe just remove this paragraph?

**Review by cmouse — CHANGES_REQUESTED** [review]

**Comment by cmouse** [comment]
can you rebase your branch?

**Comment by matejak** [comment]
I have cleaned it up and rebased.

File: source/configuration_manual/authentication/oauth2.rst [hunk 3]
```diff
@@ -152,7 +152,7 @@ Local validation
 Local validation allows validating tokens without connecting to an oauth2 server.
 This requires that key issuer supports `JWT tokens (RFC 7519) <https://tools.ietf.org/html/rfc7519>`_.
 
-You can put the validation keys into any `dictionary <https://wiki.dovecot.org/Dictionary>`.
+You can put the validation keys into any `dictionary <https://wiki.dovecot.org/Dictionary>`_.
```
Comment (cmouse) [inline source/configuration_manual/authentication/oauth2.rst:155]: Unrelated change

File: source/settings/plugin/acl.rst [hunk 4]
```diff
@@ -125,7 +125,9 @@ Global ACL file path is specified as a parameter to vfile backend in ``acl``
 setting (``/etc/dovecot/dovecot-acl`` in the above example). The file contains
 otherwise the same data as regular per-mailbox ``dovecot-acl`` files, except
 each line is prefixed by the mailbox name pattern. The pattern may contain
-``*`` and ``?`` wildcards.
+``*`` and ``?`` wildcards that do the shell-like string matching, not stopping
```
Comment (cmouse) [inline source/settings/plugin/acl.rst:128]: that do shell-string

**Review by cmouse — CHANGES_REQUESTED** [review]

File: source/configuration_manual/authentication/oauth2.rst (hunk 3 above)
Comment (matejak) [reply source/configuration_manual/authentication/oauth2.rst:155]: What to do with this one?

Comment (cmouse) [reply source/configuration_manual/authentication/oauth2.rst:155]: put it on separate commit

**Comment by matejak** [comment]
done, thanks for your remarks.

**Review by cmouse — APPROVED** [review]
